typescript/cli/src/deploy/dry-run.ts

```diff
@@ -0,0 +1,47 @@
+import { MultiProvider } from '@hyperlane-xyz/sdk';
+
+import { logYellow } from '../../logger.js';
+import { ANVIL_RPC_METHODS, getLocalProvider, setFork } from '../utils/fork.js';
+
+/**
+ * Prepares the MultiProvider for a command being dry-run
+ * @param multiProvider the MultiProvider to be prepared
+ * @param chains the chain selection passed-in by the user
+ */
+export async function prepareMultiProviderForDryRun(
+  multiProvider: MultiProvider,
+  chains: string[],
+) {
+  if (chains.length != 1)
+    throw new Error('Only one chain can be present for dry-running.');
+  const chain = chains[0];
+  multiProvider = multiProvider.extendChainMetadata({
+    [chain]: { blocks: { confirmations: 0 } },
+  });
+  await setFork(multiProvider, chain);
+}
```

Comment:
if this is only dry run lets make the chain param only one?

Reply:
Agreed. Done.